src/main.rs

```diff
@@ -295,8 +295,12 @@ async fn main() {
         settings.bitcoin.network,
     )
     .unwrap();
-    let ethereum_wallet =
-        ethereum_wallet::Wallet::new(unimplemented!(), settings.ethereum.node_url).unwrap();
+    let ethereum_wallet = ethereum_wallet::Wallet::new(
+        unimplemented!(),
+        settings.ethereum.node_url,
+        dai_contract_addr,
+    )
+    .unwrap();
```

Comment:
It'd be better to use `expect()`.

Reply:
I'd say we clean that up in another iteration. 
There are several `unwraps()` to be cleaned up.